### ISSUE TITLE
Minor "Edit" menu tweak in the "Default User Settings" list sidebar

### ISSUE THREAD

**Issue by ZaLiTHkA** [Microsoft/vscode#52772]
Issue Type: <b>Feature Request</b>

hello.. (:

this may simply be a result of my current combination of "User Settings", however, after adding the following:

```
    "files.autoSave": "onWindowChange",
```

I continued making changes to settings which, naturally, kept the file in a "dirty" state as instructed. but when I use the "Edit" (icon) > "Replace In Settings" sidebar menu, I get a popup stating:

> Unable to write into user settings because the file is dirty. Please save the user settings file first and then try again.

but at *this point* here, for the action of replacing the "User Settings" version of a targeted setting, could you make it update the in-editor copy, instead of writing directly to file?

[here's a gist](https://gist.github.com/ZaLiTHkA/a10882efb40048343ba6ae9f2c8144a9) of my current `settings.json` file, in case you need it.

VS Code version: Code 1.24.1 (24f62626b222e9a8313213fb64b10d741a326288, 2018-06-13T17:51:32.889Z)
OS version: Windows_NT x64 10.0.17134

**Comment by sandy081** [comment]
If I understand correctly, you have changed your settings file and did not save it and then you used `edit` action to update a different setting? Is that correct?

**Comment by ZaLiTHkA** [comment]
that's correct @sandy081, after saving my `settings.json` the "Replace in Settings" option copied the relevant setting across to my settings file with no issue.

just to clarify: each of these settings are behaving exactly as expected, so this is not a bug... I just *feel* like the "Replace in Settings" option should update the settings value in the open editor, rather than updating the contents of the physical file. :)

**Comment by sandy081** [comment]
You mean udpate without saving the file?

**Comment by ZaLiTHkA** [comment]
yup, exactly. :)

**Comment by sandy081** [comment]
May I know what is your reasoning behind not saving the file?

**Comment by ZaLiTHkA** [comment]
there's no special reason for it, that was just a side effect of using `"files.autoSave": "onWindowChange"`, but I set *that* option in order to make changes in multiple files, then simply <kbd>alt+tab</kbd> to my browser and have *all* files save automatically.

**Comment by sandy081** [comment]
Current behaviour is expected so closing this.

**Comment by ZaLiTHkA** [comment]
I'm confused... I didn't log a bug report with current behaviour that was *unexpected*, I logged a feature request to talk about the viability of *changing* the current behaviour.

>Current behaviour is expected so closing this.

so does this mean you're not willing to entertain that idea? must be honest, that mentality really sucks. :(

**Comment by sandy081** [comment]
@ZaLiTHkA Sorry for not being clear. I did not close it for being expected behaviour but also because there is no valid use case to support it. I am sorry that we do not add features just for no reason. I hope you understand.

Thanks

**Comment by ZaLiTHkA** [comment]
to be honest, I don't really understand. I think it's safer all-round to update the contents of a file that's currently loaded in the editor, rather than directly on disk, especially when the menu item that carries out the action is only visible once the file (in this case, `settings.json`) has been opened. you were focusing on *how* I discovered this behaviour, rather than the behaviour itself, as though whatever steps I took were the core of the matter.

but either way, all I can do is ask, which I did. so I guess that's that.
